### DISCUSSION TITLE
Prepopulate Isar database

### DISCUSSION THREAD

**Discussion by Tembocs** [isar/isar#358] [Help]
Is there a way to prepopulate Isar database with some data that can be used during development as well as after release (in production)?

**Comment by simc** [comment]
Yes, collections have an `importJson` method. It's the preferred way to populate the database.

**Reply by FallenAngel97** [reply]
@simc , sorry for necrobumping, but I haven't found anything about it in README and on official site. 
Can you please elaborate on this, is this being executed every time of application start? Can we rely on the fact, that it is a one-time operation, so it will be skipped on the subsequent launches of application?
I was searching for something in Isar, like "onCreate" in SQFlite, and as I understand, I need to write something like
`isar.collection<RequiredCollection>().importJson([]);`
Thank you so much for the library!

**Reply by simc** [reply]
Yeah the import happens immediately and only once (as long as you call it only once)

**Reply by FallenAngel97** [reply]
Once per application launch or once per application lifetime? 

```dart
void main() {
  final isar = Isar.open([RequiredSchema]);
  isar!.collection<RequiredCollection>().importJson([]); // <- this line will not be executed on the second application launch? 
  runApp(const MyApp());
}
```
I just think, should I check, that it will overwrite my data, or it will be unnecessary. 
Also, what do you think, maybe it's good to add to the FAQ or README?

**Reply by sunilguptasg** [reply]
simc said '(as long as you call it only once)'. It is a generic importJson method, so it will execute every time it is called. You could do a select, check the dB is empty, and execute the 'initial' import...

**Reply by FallenAngel97** [reply]
Thank you. I was thinking about wrapping it in "Shared preferences", and setting there a flag, so you just confirmed my expectations are correct :)

**Comment by sunilguptasg** [comment]
Shared preferences can be deleted, and I would not rely on these values completely. It's easier to check if the dB is empty.
